src/static/browser-core.d.ts

```diff
@@ -1 +1,20 @@
-declare module 'browser-core';
+declare module 'browser-core' {
+
+  interface ICliqzModuleBackground {
+  }
+
+  interface ICliqzModule {
+    name: string;
+    background: ICliqzModuleBackground;
+  }
+
+  interface IStringMap<T> {
+    [x: string]: T;
+  }
```

Comment:
Is it really needed to declare a specific interface for that? I guess it could be inlined in the `App` type directly.

Reply:
have copied it from some example. I'm still experimenting with the syntax. 

What I would really want to see here is something like:
```ts
type moduleName = 'antitracking' | 'human-web' | 'search';

type Modules {
  [name: moduleName]: ICliqzModule;
}
```

but it does not want work to work that way..

Reply:
I think what you're trying to do is called `dependent types` :P And I'm pretty sure typescript does not support it. You can only describe the type, not the values.

Reply:
Maybe we should think about how we would implement the backgrounds in the extension if we were to write the code in typescript, then the declaration for browser-core should be easier to write. Should it be a class with inheritance? Should it just be an interface with `actions`, `events`, `init` and `unload`? (`actions` and `events` begin a mapping from name to function). What do you think?